DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/Readme.md

```diff
@@ -0,0 +1,287 @@
+>
+> Before attempting to use this design, you should ensure the following directories are in your `PATH` environment variable:
+>
+> ```
+> <QUARTUS_INSTALL_DIR>/qsys/bin
+> <QUARTUS_INSTALL_DIR>/questa_fe/bin
+> ```
+>
+> :warning: Make sure you add the device files associated with the FPGA that you are targeting to your Intel® Quartus® Prime installation.
+
+## Overview of NIOS V Design
+
+The design has the following directory structure after you compile it:
+
+| Directory          | Description
+|--------------------|------------
+|`.qsys_edit        `| (generated directory) Temporary Platform Designer files
+|`DNI/              `| (generated directory) Files generated by Intel® Quartus® Prime 
+|`ip/               `| .ip files for all IP inside platform designer system
+|`kernels/          `| Directory containing oneAPI kernels (oneAPI kernel sources should be placed here)
+|`pd_system/        `| (generated directory) RTL comprising the Platform Designer system 
+|`pd_system_tb/     `| (generated directory) Platform Designer test bench system that instantiates pd_system.qsys and wraps it with BFMs
+|`qdb               `| (generated directory) Files generated by Intel® Quartus® Prime during synthesis
+|`simulation_files/ `| Scripts to automate co-simulating hardware and NIOS V software
+|`software/         `| Nios V sample software with build script (application and BSP will generated from build script)
+|`build_and_sim_testsystem.tcl`| Script to run the steps in the [build steps](#build-the-niosv-reference-design) below
+|`components.ipx    `| IP search path file that extends the search path to the /custom_ip and /kernels directories
+|`pd_system.qsys    `| Platform Designer system (including Nios V, JTAG UART, 1MB of on-chip RAM)
+|`pd_system.qsys.legacy`| (generated file) Legacy Platform Designer file
+|`test_system.qpf   `| Quartus project file
+|`test_system.qsf   `| Quartus settings file
+
+### Hardware Overview
+
+The NIOS V soft CPU is connected to the other IPs in this design using Platform Designer. This design is stored in the file `pd_system.qsys`.
+
+![](assets/platform-designer_top.png)
+
+There are three important IPs in this system:
+
+1. **On-chip Memory** This on-chip RAM is shared between the Nios V/g soft CPU and the simple DMA accelerator FPGA IP produced with the Intel® oneAPI DPC++/C++ Compiler.
+
+2. **Nios V/g soft CPU** This soft CPU executes a simple test program to exercise the simple DMA accelerator FPGA IP produced with the Intel® oneAPI DPC++/C++ Compiler.
+
+3. **Simple DMA Accelerator** This FPGA IP is produced with the Intel® oneAPI DPC++/C++ Compiler
+
+Although FPGA IP produced with the Intel® oneAPI DPC++/C++ Compiler does not support AXI4 interfaces, Platform Designer is able to generate a bridge for you. Click `System` -> `Show System with Platform Designer Interconnect` to see how the Platform Designer interconnect adapts your component's Avalon memory-mapped interfaces to AXI4 interfaces.
```

Comment:
Platform Designer doesn't build bridges for this, it just natively supports Avalon and AXI which is why there is one interconnect servicing both AXI and Avalon traffic in the screenshot.  But at this point in the readme, the user hasn't added the accelerator to the Platform Designer system, so showing them the interconnect view here might be too early.  I don't think showing them the interconnect view is adding much (power users don't even look at it much), just state that Platform designer is capable of integrating Avalon IP (like the oneAPI kernel) and AXI IP (like Nios V) without users needing to manually adapter them with bridges.

Reply:
I'll remove the interconnect view then. I don't want this to be too long.